src/imitation/algorithms/preference_comparisons.py

```diff
@@ -905,8 +927,30 @@ def __init__(
         self.optim = th.optim.AdamW(
             self._model.parameters(),
             lr=lr,
-            weight_decay=weight_decay,
         )
+        self.seed = seed
+        self.val_split = val_split
+
+        # TODO(juan) this could accept an arbitrary regularizer function
```

Comment:
I think we probably want to do that? It seems weird that we implement `L2Regularizer` etc but it's not used anywhere in our current codebase.

Reply:
I added that. It seems excessive to add a protocol and a wrapper (or split initialization of the class into two through separate methods) to first initialize the class with its appropriate regularization algorithm parameters (e.g. "p" in the Lp norm) and then another to pass python objects e.g. optimizer and logger instance, which are only accessible within the RewardTrainer. So we just pass a type as an argument, which isn't great but fairly usable and simpler IMO.